### PR TITLE
fix: publish unknown otp option

### LINKED ISSUE #6384: `pnpm publish --otp` works in 8.1.0 but no longer works in 8.2.0
### pnpm version:
8.2.0

### Code to reproduce the issue:
```
mkdir test && cd test
pnpm init
git init && git add . && git commit -m 'repo'
pnpm publish --otp=123456
```

### Expected behavior:
```
$ tea +pnpm.io@8.1.0 pnpm publish --otp=123456
npm notice
npm notice 📦  test@1.0.0
npm notice === Tarball Contents ===
npm notice 217B package.json
npm notice === Tarball Details ===
npm notice name:          test
npm notice version:       1.0.0
npm notice filename:      test-1.0.0.tgz
npm notice package size:  247 B
npm notice unpacked size: 217 B
npm notice shasum:        2a3f34a68f00cdad13702bd2f9b64f9b142ac66d
npm notice integrity:     sha512-6TPnR9sdbjxxy[...]gqqoZqCr139Rg==
npm notice total files:   1
npm notice
npm notice Publishing to https://registry.npmjs.org/ with tag latest and default access
This operation requires a one-time password.
Enter OTP:
```

### Actual behavior:
```
$ tea +pnpm.io@8.2.0 pnpm publish --otp=123456
 ERROR   ERROR  Unknown option: 'otp'
For help, run: pnpm help publish
```

### Additional information:

 - `node -v` prints: v18.4.1
 - Windows, macOS, or Linux?: macOS
 - The `tea` command is from https://tea.xyz/ that makes it easy to run different versions of a CLI

### PR DESCRIPTION
**Pull request by await-ovo**
Fix #6384 

Also add override `"yaml@<2.2.2": ">=2.2.2"` to `package.json` to fix audit vulnerabilities.

### CODE REVIEW

File: releasing/plugin-commands-publishing/src/publish.ts [hunk 1]
```diff
@@ -36,6 +35,7 @@ export function rcOptionsTypes () {
 export function cliOptionsTypes () {
   return {
     ...rcOptionsTypes(),
```
Comment (zkochan) [inline releasing/plugin-commands-publishing/src/publish.ts]: wasn't it coming from here?

Comment (await-ovo) [reply releasing/plugin-commands-publishing/src/publish.ts]: It seems that the object returned by `rcOptionsType` does not contains `otp` property.

Comment (await-ovo) [reply releasing/plugin-commands-publishing/src/publish.ts]: Well, I found that there is[ `otp` in the git repository](https://github.com/pnpm/npm-conf/blob/main/lib/types.js#L84), but the actual installation is not(https://www.npmjs.com/package/@pnpm/npm-conf?activeTab=code).

Is there something wrong with the `@pnpm/npm-conf@2.1.1` release ?

Comment (zkochan) [reply releasing/plugin-commands-publishing/src/publish.ts]: I have no idea but for sure it should come from there.

Comment (await-ovo) [reply releasing/plugin-commands-publishing/src/publish.ts]: Could we try to release a new version of `@pnpm/npm-conf`?

 `publish --provenance` also encountered an unknown option error, I added it to `@pnpm/npm-conf` in [ this PR](https://github.com/pnpm/npm-conf/pull/9).

**Pull request merged**